### PR TITLE
Update Go Tutorial

### PR DESCRIPTION
**Pull request by khwj**
Summary:
- Update project layout to be more consistent with current version of `gqlgen`
- Fix typo

### CODE REVIEW

**Comment by Glyphack** [comment]
Hey @khwj , 
Thanks for taking time to improve graphql tutorial. Once you finished with changes you can notify me for a review.
GoodLuck

**Comment by khwj** [comment]
Hey @Glyphack,

Thanks for the tutorial. For this PR, passing Netlify check might probably be it. I have a few items in mind for improvement on a separate PR. 

Maybe you could share what you’re planning to change and we can somehow collaborate?

Cheers.

**Comment by Glyphack** [comment]
Netlify is having trouble for preview and I will fix it today. I guess there is nothing wrong with your merge reuqest.
So after fixing Netlify preview I update the [tutorial repo](https://github.com/howtographql/graphql-golang) with new contents then we can merge this PR.

**Comment by Glyphack** [comment]
Hi @khwj just wanted to let you know that deploy preview is fixed you may pull the master.

File: content/backend/graphql-go/2-queries.md [hunk 1]
```diff
@@ -27,18 +27,23 @@ Let's make a dummy response for this function, for now.
...
 ```go
-func (r *queryResolver) Links(ctx context.Context) ([]*Link, error) {
-	var links []*Link
-	links = append(links, &Link{Title: "our dummy link", Address: "https://address.org", User: &User{Username: "admin"}})
+func (r *queryResolver) Links(ctx context.Context) ([]*model.Link, error) {
+  var links []*model.Link
+  dummyLink := model.Link{
+    Title: "our dummy link",
+    Address: "https://address.org",
+    User: &model.User{Username: "admin"},
```
Comment (Glyphack) [inline content/backend/graphql-go/2-queries.md:37]: `model.User` has Name attribute not Username please fix this too.

**Comment by Glyphack** [comment]
In new gqlgen version some namespaces changed and main function does not match the latest version.
https://github.com/howtographql/howtographql/pull/1094/files#diff-47904b7ddd9e5ff40dd10b2f0aa97d1eR157

I guess you can change it to:
```go
func main() {
	port := os.Getenv("PORT")
	if port == "" {
		port = defaultPort
	}

	database.InitDB()
	database.Migration()
        // Other codes
```

**Comment by Glyphack** [comment]
Hi @khwj  are you still interested in this PR?

**Comment by ashwamegh** [comment]
Hi @Glyphack and @khwj I have updated the Go tutorial, I found a lot of issues in it. Please do have a look at my PR #1136 👍

**Comment by Glyphack** [comment]
This PR can be closed I updated gqlgen version.

**Pull request closed without merging**